py3status/modules/diskdata.py

```diff
@@ -87,17 +105,73 @@ def post_config_hook(self):
             value - value (float)
             unit - unit (string)
         """
-        self.last_diskstats = self._get_diskstats(self.disk)
+        types_info = {
+            "disks": {"major": [3, 8, 22]},
+            "raid": {"major": [9]},
+            "cdrom": {
+                "major": [
+                    11,
+                    15,
+                    16,
+                    17,
+                    18,
+                    20,
+                    23,
+                    24,
+                    25,
+                    26,
+                    27,
+                    28,
+                    29,
+                    30,
+                    32,
+                    33,
+                    34,
+                ]
+            },
+            "mmc": {"major": [31]},
+        }
+
+        self._disks = {}
+
+        if not self.disk:
+            allowed_types = []
+            for _type in self.types.split():
+                allowed_types.extend(types_info[_type]["major"])
```

Comment:
It doesn't matter, but can you use `allowed_types = set()` and replace `extend` with `update`? Unique like a unicorn.

Reply:
good idea, done that.